src/SapNwRfc/Pooling/SapPooledConnection.cs

```diff
@@ -48,6 +48,46 @@ public void Dispose()
             _disposed = true;
         }
 
+        /// <inheritdoc cref="ISapPooledConnection"/>
+        public ISapTypeMetadata GetTypeMetadata(string typeName, CancellationToken cancellationToken = default)
+        {
+            _connection = _connection ?? _pool.GetConnection(cancellationToken);
```

Comment:
Hm, https://github.com/huysentruitw/SapNwRfc/pull/53/files#r776654518 used it outside of the try catch and there is even a test for it. But I will remove all changes from here regardless because of https://github.com/huysentruitw/SapNwRfc/pull/53/files#r803394569